### PR TITLE
Release 1.0.0

### PR DESCRIPTION
**Pull request by MarekSuchanek**

### CODE REVIEW

**Comment by kburger** [comment]
Since it's a bit much to go through at one, I'll try to read and comment the parts that are important to me. Will start going through it asap.

File: CHANGELOG.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
+# Changelog
+
+All notable changes to this project will be documented in this file.
+
+The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
+and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [Unreleased]
+
+## 1.0.0
```
Comment (kburger) [inline CHANGELOG.md]: You can make these headers easily linkable by putting a section like this at the bottom of the document:

```md
[Unreleased]: /../../compare/v1.0.0...develop
[1.0.0]: /../../tree/v1.0.0
```

File: pom.xml [hunk 2]
```diff
@@ -0,0 +1,341 @@
...
+            <groupId>org.eclipse.rdf4j</groupId>
+            <artifactId>rdf4j-rio-api</artifactId>
+            <version>${rdf4j.version}</version>
+        </dependency>
+
+        <dependency>
+            <groupId>org.eclipse.rdf4j</groupId>
+            <artifactId>rdf4j-rio-turtle</artifactId>
+            <version>${rdf4j.version}</version>
+        </dependency>
```
Comment (kburger) [inline pom.xml]: Could be `runtime` scoped, but it's a very minor detail.

File: src/main/java/solutions/fairdata/openrefine/metadata/fdp/FairDataPointClient.java [hunk 3]
```diff
@@ -0,0 +1,502 @@
...
+            HttpURLConnection.HTTP_SEE_OTHER,
+            307,  // HTTP Temporary Redirect
+            308   // HTTP Permanent Redirect
+    ));
+
+    private static final String CATALOG_PART = "/catalog";
+    private static final String DATASET_PART = "/dataset";
+    private static final String DISTRIBUTION_PART = "/distribution";
+    private static final String SPEC_PART = "/spec";
+    private static final String AUTH_PART = "/tokens";;
```
Comment (kburger) [inline src/main/java/solutions/fairdata/openrefine/metadata/fdp/FairDataPointClient.java]: Very very minor: duplicated trailing semicolon.

File: src/main/java/solutions/fairdata/openrefine/metadata/fdp/FairDataPointClient.java [hunk 4]
```diff
@@ -0,0 +1,502 @@
...
+        MetadataTransformerUtils.setTimestamps(datasetMetadata);
+        String metadata = MetadataUtils.getString(datasetMetadata, RDFFormat.TURTLE);
+
+        BufferedWriter bw = new BufferedWriter(new OutputStreamWriter(conn.getOutputStream(), StandardCharsets.UTF_8));
+        bw.write(metadata);
+        bw.flush();
+        bw.close();
+
+        if(conn.getResponseCode() == HttpURLConnection.HTTP_CREATED) {
+            String actualURI = conn.getHeaderField("Location");
```
Comment (kburger) [inline src/main/java/solutions/fairdata/openrefine/metadata/fdp/FairDataPointClient.java]: Do you have any constants on the classpath to re-use for header names? Guava, or a commons lib?

File: src/main/java/solutions/fairdata/openrefine/metadata/fdp/FairDataPointClient.java [hunk 5]
```diff
@@ -0,0 +1,502 @@
...
+     *
+     * @param conn connection with FAIR Data Point
+     * @param uri base URI to be retrieved
+     * @return list of statements for given URI
+     * @throws IOException in case of communication error with FDP
+     */
+    private ArrayList<Statement> parseStatements(HttpURLConnection conn, String uri) throws IOException {
+        TurtleParser parser = new TurtleParser();
+        StatementCollector collector = new StatementCollector();
+        parser.setRDFHandler(collector);
```
Comment (kburger) [inline src/main/java/solutions/fairdata/openrefine/metadata/fdp/FairDataPointClient.java]: Could be replaced with the `RDFParser parser = Rio.createParser(RDFFormat.TURTLE);`, but the end result would be exactly the same. 🤷‍♂

File: src/main/java/solutions/fairdata/openrefine/metadata/storage/TripleStoreHTTPStorage.java [hunk 6]
```diff
@@ -0,0 +1,90 @@
...
+import java.util.HashMap;
+
+public class TripleStoreHTTPStorage extends Storage {
+
+    public static final String TYPE = "tripleStoreHTTP";
+    private static final HashMap<String, RDFFormat> formats = new HashMap<>();
+    private final HTTPRepository repository;
+
+    static {
+        formats.put("application/x-turtle", RDFFormat.TURTLE);
```
Comment (kburger) [inline src/main/java/solutions/fairdata/openrefine/metadata/storage/TripleStoreHTTPStorage.java]: Could be replaced with `RDFFormat.getMIMETypes()`, but I'm not sure if you are going for specific MIME types here.

File: CHANGELOG.md (hunk 1 above)
Comment (MarekSuchanek) [reply CHANGELOG.md]: That's a great idea! 👍

Comment (kburger) [reply CHANGELOG.md]: It's based on the `markdown` example on https://keepachangelog.com/en/1.0.0/ by the way, but there they use the absolute URLs, where you can use relative URLs using the `/../../` trick.

**Comment by MarekSuchanek** [comment]
> Added some minor comments.

Thank you, I used all the advice in https://github.com/FAIRDataTeam/OpenRefine-metadata-extension/pull/28/commits/02ef84d01b6caac1b84431c2edd645fff6a722f4

**Pull request merged**